rwengine/src/render/ObjectRenderer.cpp

```diff
@@ -349,24 +349,24 @@ void ObjectRenderer::buildRenderList(GameObject* object, RenderList& outList) {
     // Right now specialized on each object type
     switch (object->type()) {
         case GameObject::Instance:
-            renderInstance(static_cast<InstanceObject*>(object), outList);
+            renderInstance(dynamic_cast<InstanceObject*>(object), outList);
```

Comment:
seems more like a design flaw, assuming there is a renderInstance for each type, then probably InstanceObject should have a renderInstance() which is passed an ObjectRenderer who can then call it without a cast.